app/helpers/admin/injection_helper.rb

```diff
@@ -184,6 +184,12 @@ def admin_inject_spree_api_key
                        json: "'#{@spree_api_key}'" }
     end
 
+    def admin_inject_available_units
```

Comment:
I'd prefer to have this as an API endpoint of configurations and not as injected data in the dom. What do you think?

Reply:
we would have a preferences api endpoint where we could expose quite a few of them including this one :tada:

Reply:
Let me think about that some more... :) I can see both sides. On the one hand it's not so elegant to inject a string into the DOM in a bunch of places like we're doing now. And we will want to make this an API endpoint eventually anyway (since we want to make pretty much everything an API endpoint, right?)

I think my hesitation in having the front end call it as an endpoint is that it means an additional https call to the server, which 1) slows things down and 2) could make the code more complicated, since the methods that depend on knowing the value of `available_units` now have to wait for that request to return... but maybe we handle similar situations elsewhere and have a good pattern to follow for it. 

I think this tension will come up a lot, between having the front end javascript strictly use the API vs. having some data already available to it in the DOM. 

I'm going to sleep on it for now :)

Reply:
yes, you comment is spot on.
My opinion is very simple: we must use the API approach in 95% of the cases if we ever want to have an useful API. Otherwise another 2/3 years will pass and not much api evolution will happen, like what happened in the last 2 years I have been around.

Reply:
Yes, I like keeping it disciplined. And I think this case is not so exceptional that it should be in the 5% of times when we don't use the API; it'll be a necessary endpoint if some client were trying to recreate the front end using only the API, so I think we should create it and eat our own dogfood.